src/main/java/com/minecolonies/coremod/util/BarbarianUtils.java

```diff
@@ -0,0 +1,193 @@
+package com.minecolonies.coremod.util;
+
+import com.minecolonies.api.configuration.Configurations;
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.entity.ai.mobs.EntityArcherBarbarian;
+import com.minecolonies.coremod.entity.ai.mobs.EntityBarbarian;
+import com.minecolonies.coremod.entity.ai.mobs.EntityChiefBarbarian;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityList;
+import net.minecraft.init.Items;
+import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.world.World;
+
+import java.util.stream.IntStream;
+
+/**
+ * Utils for The Barbarian Raid Event
+ */
+public class BarbarianUtils
+{
+    /**
+     * ResourceLocations for barbarians
+     */
+    /* default */ public static ResourceLocation barbarian = EntityList.getKey(EntityBarbarian.class);
+    /* default */ public static ResourceLocation archer    = EntityList.getKey(EntityArcherBarbarian.class);
+    /* default */ public static ResourceLocation chief     = EntityList.getKey(EntityChiefBarbarian.class);
+
+    final static int LEVEL_AT_WHICH_TO_NOT_TRIGGER_RAID = 1;
+    final static int MAX_SIZE                           = Configurations.maxBarbarianHordeSize;
+
+    public static void doRaid(final World world, final int level, final Colony colony)
+    {
+        eventRaid(world, level, colony);
+    }
+
+    /**
+     * Commences a Raid event in a specific colony, with a specific RaidLevel
+     */
+
+    private static void eventRaid(final World raidingWorld, final int level, final Colony colony)
+    {
+        if (level == LEVEL_AT_WHICH_TO_NOT_TRIGGER_RAID)
+        {
+            return;
+        }
+
+        int numberOfBarbarians = level;
+        int numberOfArcherBarbarians = (int) (0.5 * level);
+        int numberOfChiefBarbarians = (int) (0.1 * level);
+
+        int hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+
+        if (hordeTotal > 40 && MAX_SIZE == 40 && hordeTotal > MAX_SIZE)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 40 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 40 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)